src/components/Tooltip.vue

```diff
@@ -1,31 +1,31 @@
 <template>
     <span class="tooltip"
-        :class="{ active: tooltipActive }"
+        :class="[tooltipPosition, {
+            shown: isShown,
+            'transition-position': transitionPosition,
+            'inverse-theme': theme === constructor.Themes.INVERSE,
+        }]"
         @mouseenter="mouseOver(true)"
         @mouseleave="mouseOver(false)"
     >
         <a href="javascript:void(0);"
-            ref="tooltipIcon"
-            @focus.stop="toggleTooltip"
-            @blur.stop="tooltipToggled ? toggleTooltip() : ''"
-            :class="{
-                top: tooltipPosition === 'top',
-                bottom: tooltipPosition === 'bottom',
-            }">
-            <slot name="icon">
+            ref="tooltipTrigger"
+            @focus.stop="show"
+            @blur.stop="hide"
+            :tabindex="disabled ? -1 : false"
+            class="trigger">
+            <slot name="trigger">
                 <AlertTriangleIcon class="nq-orange" />
             </slot>
         </a>
-        <div ref="tooltipBox"
-            class="tooltip-box"
-            :style="styles"
-            :class="{
-                active: tooltipActive,
-                top: tooltipPosition === 'top',
-                bottom: tooltipPosition === 'bottom',
-            }">
-            <slot></slot>
-        </div>
+        <transition>
```

Comment:
I think we should name the transition.

Reply:
Not necessary in my opinion but I set a name.